python_src/src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -0,0 +1,442 @@
+from typing import List
+import binascii
+
+import numpy
+import pandas
+import sqlalchemy as sa
+
+from lamp_py.postgres.postgres_utils import DatabaseManager
+from lamp_py.postgres.postgres_schema import (
+    VehicleTrips,
+    TempHashCompare,
+)
+from lamp_py.runtime_utils.process_logger import ProcessLogger
+from .l1_cte_statements import (
+    get_rt_trips_cte,
+    get_static_trips_cte,
+)
+
+
+def insert_trips_hash_columns(
+    db_manager: DatabaseManager, events: pandas.DataFrame
+) -> None:
+    """
+    Load new 'trip_hash' records and columns into RDS
+
+    This guarantees that all events represented in the events dataframe will have
+    matching trips data in the VehicleTrips table
+    """
+    insert_columns = [
+        "trip_hash",
+        "fk_static_timestamp",
+        "direction_id",
+        "route_id",
+        "start_date",
+        "start_time",
+        "vehicle_id",
+    ]
+    insert_events = events.loc[
+        events["do_insert"], insert_columns
+    ].drop_duplicates(subset="trip_hash")
+
+    new_trip_hash_bytes = insert_events["trip_hash"]
+
+    db_manager.truncate_table(TempHashCompare)
+    db_manager.execute_with_data(
+        sa.insert(TempHashCompare.__table__),
+        new_trip_hash_bytes.to_frame("trip_stop_hash"),
+    )
+
+    db_trip_hashes = db_manager.select_as_dataframe(
+        sa.select(
+            VehicleTrips.trip_hash,
+        ).join(
+            TempHashCompare,
+            TempHashCompare.trip_stop_hash == VehicleTrips.trip_hash,
+        )
+    )
+
+    insert_events["trip_hash"] = (
+        insert_events["trip_hash"].apply(binascii.hexlify).str.decode("utf-8")
+    )
+
+    if db_trip_hashes.shape[0] == 0:
+        db_trip_hashes = pandas.DataFrame(columns=["trip_hash"])
+    else:
+        db_trip_hashes["trip_hash"] = (
+            db_trip_hashes["trip_hash"]
+            .apply(binascii.hexlify)
+            .str.decode("utf-8")
+        )
+
+    # insert_events are records that don't have an existing trip_hash in the VehicleTrips table
+    insert_events = insert_events.merge(
+        db_trip_hashes, how="outer", on="trip_hash", indicator=True
+    ).query('_merge=="left_only"')
+
+    insert_events = insert_events.drop(columns=["_merge"])
+
+    insert_events["trip_hash"] = insert_events["trip_hash"].str.decode("hex")
+
+    if insert_events.shape[0] > 0:
+        db_manager.execute_with_data(
+            sa.insert(VehicleTrips.__table__),
+            insert_events,
+        )
+
+
+# pylint: disable=R0914
+# pylint too many local variables (more than 15)
+def process_trips_table(
+    db_manager: DatabaseManager,
+    seed_start_date: int,
+    seed_timestamps: List[int],
+) -> None:
+    """
+    processs updates to trips table
+
+    generate new trips table information
+    """
+    process_logger = ProcessLogger("l1_rt_trips_table_loader")
+    process_logger.log_start()
+
```

Comment:
All of the business logic in `process_trips_table` is basically the same as before, except this will only ever result in UPDATES to the trip table and never INSERT any new trips records.